internal/commands/vaultsecrets/integrations/displayer.go

```diff
@@ -93,14 +104,181 @@ func (m *mongodbDisplayer) previewMongoDBIntegrationsPayload() any {
 }
 
 func (m *mongodbDisplayer) FieldTemplates() []format.Field {
-	return []format.Field{
+	fields := []format.Field{
 		{
 			Name:        "Integration Name",
-			ValueFormat: "{{ .IntegrationName }}",
+			ValueFormat: "{{ .Name }}",
 		},
+	}
+
+	if m.single {
+		return append(fields, []format.Field{
+			{
+				Name:        "API Public Key",
+				ValueFormat: "{{ .StaticCredentialDetails.APIPublicKey }}",
+			},
+		}...)
+	} else {
+		return fields
+	}
+}
+
+type awsDisplayer struct {
+	previewAwsIntegrations []*preview_models.Secrets20231128AwsIntegration
+
+	single bool
+}
+
+func newAwsDisplayer(single bool, integrations ...*preview_models.Secrets20231128AwsIntegration) *awsDisplayer {
+	return &awsDisplayer{
+		previewAwsIntegrations: integrations,
+		single:                 single,
+	}
+}
+
+func (a *awsDisplayer) DefaultFormat() format.Format {
+	return format.Table
+}
+
+func (a *awsDisplayer) Payload() any {
+
+	if a.previewAwsIntegrations != nil {
+		return a.previewAwsIntegrationsPayload()
+	}
+
+	return nil
+}
+
+func (a *awsDisplayer) previewAwsIntegrationsPayload() any {
+	if a.single {
+		if len(a.previewAwsIntegrations) != 1 {
+			return nil
+		}
+		return a.previewAwsIntegrations[0]
+	}
+	return a.previewAwsIntegrations
+}
+
+func (a *awsDisplayer) FieldTemplates() []format.Field {
+	fields := []format.Field{
 		{
-			Name:        "API Public Key",
-			ValueFormat: "{{ .MongodbAPIPublicKey }}",
+			Name:        "Integration Name",
+			ValueFormat: "{{ .Name }}",
+		},
+	}
+
+	if a.single {
+		return append(fields, []format.Field{
+			{
+				Name:        "Audience",
+				ValueFormat: "{{ .FederatedWorkloadIdentity.Audience }}",
```

Comment:
I believe we also return Role ARN in the response, not sure if you want to also display it.